### PR TITLE
refactor(lib): add common utils

### PR DESCRIPTION
**Pull request by chronolaw**
KAG-5215

### CODE REVIEW

**Review by ADD-SP — APPROVED** [review]

File: lualib/resty/events/utils.lua [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+
+local function is_closed(err)
+    return err and (str_sub(err, -6) == "closed" or
+                    str_sub(err, -11) == "broken pipe")
+end
+
+
+local function get_worker_id()
+    return ngx_worker_id() or -1
```
Comment (outsinre) [inline lualib/resty/events/utils.lua:17]: Better add a comment like

> `-1` represents priviledged worker

File: lualib/resty/events/init.lua [hunk 2]
```diff
@@ -1,22 +1,26 @@
...
+local get_worker_id = utils.get_worker_id
+
 
 local type = type
 local setmetatable = setmetatable
 local str_sub = string.sub
 
-local worker_count = ngx.worker.count()
+
+local worker_count = ngx.worker.count() -- luacheck: ignore
```
Comment (outsinre) [inline lualib/resty/events/init.lua:15]: Can we put this in the uitls?

**Comment by outsinre** [comment]
LGTM with misc comments.

**Review by outsinre — APPROVED** [review]

File: lualib/resty/events/utils.lua (hunk 1 above)
Comment (chronolaw) [reply lualib/resty/events/utils.lua]: done, thanks.

File: lualib/resty/events/init.lua (hunk 2 above)
Comment (chronolaw) [reply lualib/resty/events/init.lua]: -1 represents priviledged worker

**Review by outsinre — DISMISSED** [review]

File: lualib/resty/events/utils.lua [hunk 3]
```diff
@@ -0,0 +1,43 @@
+local str_sub = string.sub
+
+
+local ngx = ngx -- luacheck: ignore
+local ngx_worker_id = ngx.worker.id
+local ngx_worker_count = ngx.worker.count()
```
Comment (outsinre) [inline lualib/resty/events/utils.lua:6]:
```suggestion
local ngx_worker_count = ngx.worker.count
```

Comment (chronolaw) [reply lualib/resty/events/utils.lua]: fixed, thanks.

**Review by outsinre — APPROVED** [review]

**Review by bungle — APPROVED** [review]

**Pull request merged**